iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -469,12 +474,34 @@ private Long getSnapshotId(Table table, String branchName) {
   /**
    * Creates and commits an Iceberg change with the provided data and delete files.
    * If there are no delete files then an Iceberg 'append' is created, otherwise Iceberg 'overwrite' is created.
-   * @param table The table we are changing
-   * @param startTime The start time of the commit - used only for logging
-   * @param results The object containing the new files we would like to add to the table
+   *
+   * @param table        The table we are changing
+   * @param startTime    The start time of the commit - used only for logging
+   * @param results      The object containing the new files we would like to add to the table
+   * @param isCompaction Flag that indicates whether or not the operation is done as part of compaction
    */
   private void commitWrite(Table table, String branchName, Long snapshotId, long startTime,
-      FilesForCommit results, Operation operation) {
+                           FilesForCommit results, Operation operation, boolean isCompaction) {
+
+    if (isCompaction) {
```

Comment:
could you please extract this logic into existing commitOverwrite method and in case of MAJOR compaction use enum REWRITE_ALL_PARTITIONS.
In future we might map new DDL operation `OPTIMIZE TABLE REWRITE DATA` to that logic in order to fix issues with partition evolution + rewrite small files.

Reply:
````
enum RewriteData {DEFAULT, ALL_PARTITIONS}
````
pass this instead of isCompaction()

Reply:
Done